### PR TITLE
Store a reference to the tooltip/popover on insert

### LINKED ISSUE #765: JS error occurs when a tooltip/popover is hidden
Introduced by https://github.com/rstudio/bslib/pull/749 (via https://github.com/twbs/bootstrap/pull/37235). With that recent change, a JS error occurs when a tooltip/popover is hidden

```r
tooltip("Hover me", "Tooltip message")
```

```
Uncaught Error: Failed to find the popover's DOM element. Please report this bug.
```

### PR DESCRIPTION
**Pull request by cpsievert**
Closes #765

### CODE REVIEW

File: srcts/src/components/popover.ts [hunk 1]
```diff
@@ -282,15 +286,10 @@ export class BslibPopover extends BslibElement {
...
   // element.
   private _restoreContent(): void {
-    const { tip } = this.bsPopover;
-    if (!tip) {
-      throw new Error(
-        "Failed to find the popover's DOM element. Please report this bug."
-      );
-    }
-    const body = tip.querySelector(".popover-body");
+    const el = this.bsPopoverEl;
```
Comment (gadenbuie) [inline srcts/src/components/popover.ts:289]: Do we unset this element at some point? Or do we just update it when a new popover is inserted?

**Review by gadenbuie — APPROVED** [review]
I'm still left with the general feeling that the references we should track in the class instance should be to elements that _we control_, i.e. the popover body and title. But I also understand that this may be a quicker and easier way to solve the current bug. I'm happy if we just make a note to think about this again if we end up in this area again.

**Comment by cpsievert** [comment]
> I'm still left with the general feeling that the references we should track in the class instance should be to elements that we control, i.e. the popover body and title.

Yea, I had another quick look into this. That approach would definitely simplify some logic and make it more robust to changes to Bootstrap. The downside though is that (I think), if we don't reach into the tooltip/popover element (to return the contents to their rightful place), then we'd open ourselves up to restoring "stale" content, which is definitely a problem for popovers

File: srcts/src/components/popover.ts (hunk 1 above)
Comment (cpsievert) [reply srcts/src/components/popover.ts:292]: Ah yes, good point. Addressed in [c0b95a9](https://github.com/rstudio/bslib/pull/766/commits/c0b95a9dee5357e107fde0b315023e82e1a645ee)

**Pull request merged**